Sync port services to incident io (#1617)
port-labs/port-docs: docs/guides/all/sync-service-entities-to-incident-io.md

---
sidebar_position: 8
displayed_sidebar: null
description: Learn how to sync your Port service entities into Incident IO catalog.
---

import Tabs from "@theme/Tabs"
import TabItem from "@theme/TabItem"
import PortTooltip from "/src/components/tooltip/tooltip.jsx"
import PortApiRegionTip from "/docs/generalTemplates/_port_region_parameter_explanation_template.md"

# Sync Port Services to Incident IO
This guide helps you set up a GitHub workflow that periodically syncs service entities from Port into Incident IO's catalog, ensuring better visibility and context for your services during incident management.

## Prerequisites

> :::info Prerequisites

Reviewer: no need for an admonition here, the "Prerequisites" header is enough.

## Create GitHub workflow

Let's go ahead and create a GitHub workflow file in a GitHub repository meant for the Incident IO integration:

- Create a GitHub repository (or use an existing one)
- Create a `.github` directory and `workflows` sub directory

Inside the `.github/workflows` directory create a file called `sync-port-services-to-incident-io.yml` with the following content:

<details>
<summary><b> GitHub workflow configuration (click to expand) </b></summary>

:::tip schedule interval
The default syncing interval is set to 2 hours. Adjust it to suit your use case
:::

```yml showLineNumbers
name: Sync Data to Incident IO
on:
  schedule:
    - cron: "0 */2 * * *" # every two hours. Adjust this value
jobs:
  sync-data:
    runs-on: ubuntu-latest
    steps:
      - name: Check out the repository
        uses: actions/checkout@v4
        

>       - name: Get Port Access Token

Author: Yes, when using Port's github-app, you don't need to get the access token since that's encapsulated when making calls to Port's API. But besides that, we don't have support for it yet

              --arg external_id "$(echo "$entity" | jq -r '.identifier')" \
              --arg name "$name" \
              --arg catalog_type_id "${{ secrets.INCIDENT_IO_CATALOG_TYPE_ID }}" \
              '{
                "aliases": [],
                "attribute_values": {
                  ($url_id): {"value": {"literal": $url}},
                  ($readme_id): {"value": {"literal": $readme}},
                  ($language_id): {"value": {"literal": $language}},
                  ($lifecycle_id): {"value": {"literal": $lifecycle}},
                  ($type_id): {"value": {"literal": $type}}
                },
                "catalog_type_id": $catalog_type_id,
                "external_id": $external_id,
                "name": $name
              }')

            echo "Sending data to API for entity $name"

            response=$(curl -i -X POST "https://api.incident.io/v2/catalog_entries" \
              -H "Authorization: Bearer ${{ secrets.INCIDENT_IO_API_KEY }}" \
              -H "Content-Type: application/json" \
              -d "$data")

            echo "Response from Incident.io API for entity $name"
          done
```

<PortApiRegionTip/>
</details>

> ## See Results

Reviewer: ```suggestion
## Results
```

---
sidebar_position: 8
displayed_sidebar: null
description: Learn how to sync your Port service entities into incident.io catalog.
---

import Tabs from "@theme/Tabs"
import TabItem from "@theme/TabItem"
import PortTooltip from "/src/components/tooltip/tooltip.jsx"
import PortApiRegionTip from "/docs/generalTemplates/_port_region_parameter_explanation_template.md"

# Sync Port Services to incident.io

> This guide helps you set up a GitHub workflow that periodically syncs service entities from Port into incident.io's catalog, ensuring better visibility and context for your services during incident management.

Reviewer: ```suggestion
This guide demonstrates how to set up a GitHub workflow that periodically syncs service entities from Port into incident.io's catalog, ensuring better visibility and context for your services during incident management.
```

              -H "Authorization: Bearer ${{ secrets.INCIDENT_IO_API_KEY }}" \
              -H "Content-Type: application/json" \
              -d "$data")

            echo "Response from Incident.io API for entity $name"
          done
```

<PortApiRegionTip/>
</details>

## Results

Once the scheduled GitHub workflow is triggered, the service entities from Port will be automatically synced into your incident.io catalog. To view the ingested catalog items:

1. Log in to your incident.io account.
2. Navigate to the **Catalog** section in the left navigation bar.
3. Search for the `Port Services` catalog type or any custom name you provided.
4. You should now see the synced service entities from Port listed under this catalog type, giving you improved visibility and context for managing incidents.

<img src="/img/guides/IncidentioServiceCatalog.png" border="1px" />

With this integration in place, you’ll have real-time access to your Port services within incident.io, streamlining your incident response and management processes.

## Limitations

While the guide enables you to sync service entities from Port to Incident IO, there are a few limitations to keep in mind:

> - Port properties such as aggregated and mirror properties cannot be directly synced to incident.io

Reviewer: ```suggestion
- Port property types such as aggregation and mirror properties cannot be directly synced to incident.io.
```

> - Incident IO can currently ingest up to **50,000 catalog items**, so keep this limit in mind when scaling your service catalog.

Reviewer: ```suggestion
- incident.io can currently ingest up to **50,000 catalog items**, so keep this limit in mind when scaling your service catalog.
```
